public/app/features/explore/state/query.test.ts

```diff
@@ -546,4 +568,187 @@ describe('reducer', () => {
       ).toBeDefined();
     });
   });
+
+  describe('log sample', () => {
```

Comment:
That's a lot of copy-pasta 🍝  Not sure if we need it since you made the logic more generic 🤔  Or maybe we could have one mock that implements both providers and merge all tests cases?

Reply:
Also if we had unified interface mentioned earlier (DataSourceWithSupplementaryQueriesSupport) we could make the test simpler.